### PR TITLE
BREAKING CHANGE: Replication profiles and dataset IDs

### LINKED ISSUE #45: [Feature] Replication Profile
UI needs to be modified to support https://github.com/application-research/delta-dm/pull/58

I think this would be best implemented as a brand new page - either underneath `Replications` or as a top-level `Replication Profiles` 

### Modifications to existing pages
- Modify the `Datasets` page, remove `Unsealed ` and  `Indexed` columns
- Modify the `Providers` page, remove the `Allowed Datasets`
- Modify the `New Dataset` modal, removing `Publish to indexer?` and `Keep unsealed copy?`
- Modify the `Providers` `Manage` menu, removing the ability to set allowed datasets

### New features
- Add a `Replication Profiles` page
- This page should behave similar to the `Providers` page- a tabular list of the different replication profiles
- This page should have a Search function similar to Providers, so the user can search/filter them by provider or dataset
- Column headings: `Dataset`, `Provider`, `Indexed`, `Unsealed`
- Add a `New replication profile` button, to facilitate adding new profiles. User must specify the dataset/provider and 2 bools (unsealed/indexed)
- Add a `Modify` button to the page, so existing profiles may be modified (node: sp id / dataset id cannot be modified, only the bool flags unsealed/indexed can be changed. If the user wants to change the sp/dataset, they should delete and create a new one)
- Add a `Delete` button to the page, so that replication profiles may be deleted by the user

- _nice to have (not required)_ - modify the `Datasets` and the `Providers` page, adding a link to the `Replication Profiles` page where the list is auto-filtered by that dataset/provider

### LINKED ISSUE #50: [Chore] Update API calls to use `dataset ID` instead of `name`
ref: https://github.com/application-research/delta-dm/pull/65 -- use this branch when testing 

The [updated API docs ](https://github.com/application-research/delta-dm/pull/65/files#diff-9eddf4dc51bf6a0125ca7fb094468ad284112270385c41cebce4f8f0a29620ab) contain the new params required.

TLDR: 
- Any request to the backend that references a `dataset name` should be changed to use the `dataset ID`
- The one exception to this is the `POST /datasets` endpoint, the request body still requires a `name` as the ID is auto-assigned by the database, i.e

```jsonc
{
	"name": "delta-test",
	"replication_quota": 6,
	"deal_duration": 540,
}
```

### PR DESCRIPTION
**Pull request by elijaharita**
Closes #50
Closes #45 

image

### CODE REVIEW

**Comment by jcace** [comment]
missed a spot - assigning wallet to dataset 


Screenshot 2023-06-01 at 4 41 33 PM

File: components/apps/ddm/forms/FormAddReplication.tsx [hunk 1]
```diff
@@ -45,6 +45,10 @@ export default function FormAddReplication(props: {
     }
   }
 
+  function formValid() {
+    return !!providerID;
```
Comment (jcace) [inline components/apps/ddm/forms/FormAddReplication.tsx:49]: btw you can also do `Boolean(providerID)`, I find it a bit cleaner but personal preference

**Review by jcace — APPROVED** [review]
lgtm
